### ISSUE TITLE
Change Request: Update `eslint:recommended` for v9.0.0

### ISSUE THREAD

**Issue by mdjermanovic** [eslint/eslint#17596]
### ESLint version

8.49.0

### What problem do you want to solve?

The following rules have been added in ESLint v8:

* `logical-assignment-operators`
* `no-constant-binary-expression`
* `no-empty-static-block`
* `no-new-native-nonconstructor`
* `no-object-constructor`
* `no-unused-private-class-members`
* `prefer-object-has-own`

### What do you think is the correct solution?

I'd propose that in v9.0.0 we add the following rules to `eslint:recommended`:

* `no-constant-binary-expression` - reports possible errors
* `no-empty-static-block` - similar to `no-empty`, which is already in `eslint:recommended`
* `no-unused-private-class-members` - reports possible errors

Note: we have a separate issue for `no-new-native-nonconstructor` (https://github.com/eslint/eslint/issues/17446)

### Participation

- [X] I am willing to submit a pull request for this change.

### Additional comments

Did I miss some rules we added in v8?

**Comment by fasttime** [comment]
Sounds good! Maybe we can reduce to scope of #17446 to just deprecating `no-new-symbol` and add `no-new-native-nonconstructor` to this issue instead, if this makes more sense.

**Comment by fasttime** [comment]
> Did I miss some rules we added in v8?

Running

```shell
git diff v7.32.0 origin/main --name-only --diff-filter=A lib/rules/*.js
```

only shows source files for the seven rules mentioned in this issue.

**Comment by nzakas** [comment]
We should also remove `no-extra-semi` and `no-mixed-spaces-and-tabs`, as they are deprecated.

**Comment by Pyrolistical** [comment]
Can we also add `no-useless-rename`? imo this should have been part of the recommended set since its inception. This isn't something that formatters prettier would fix, so no conflict.

`no-useless-rename` is helpful when a symbol is renamed.

```js
// BEFORE RENAME
const x = {foo: 42};
const {foo: bar} = x;

// AFTER RENAME
const x = {bar: 42};
const {bar: bar} = x;
```

**Comment by mdjermanovic** [comment]
`no-useless-rename` is a stylistic rule. As described in [Using eslint:recommended](https://eslint.org/docs/latest/use/configure/configuration-files#using-eslintrecommended), this configuration enables only rules that report possible problems.

**Comment by Pyrolistical** [comment]
I think `no-useless-rename` isn't a stylistic rule. It is dead code, which is similar to `no-unused-vars` which is included in the recommended set.

**Comment by nzakas** [comment]
@Pyrolistical the changes have already been decided upon so we won't be adding any other rules to `eslint:recommended` for v9.
